3-async-await/solution.js

```diff
@@ -20,16 +20,129 @@ Example:
 9. as extra challenge: add Promise.race() and Promise.any(), and try to get the idea of what happens
 */
 
-function solution() {
-    // YOUR SOLUTION GOES HERE
+const pricesAsync = require("./prices");
+const productsAsync = require("./products");
 
-    // You generate your id value here
+/**
+ * 
+ * @param {*} id positive number
+ * @returns returns either:
+ *                    - a string with the error message or 
+ *                    - an object as follow: { id: 1, price: 0.5, product: "salt" }
+ */
 
-    // You use Promise.all() here
+async function solution(id) {
+  // YOUR SOLUTION GOES HERE
+  const promiseAllSolution = async (id) => {
+    try {
+      let resultsAll = await Promise.all([pricesAsync(id), productsAsync(id)]);
+      return {
+        id,
+        price: resultsAll[0],
+        product: resultsAll[1],
+      };
```

Comment:
we can use destructuring here
```suggestion
      let [price, product] = await Promise.all([pricesAsync(id), productsAsync(id)]);
      return {
        id,
        price,
        product,
      };
```

Reply:
Awesome! promise.all returns directly the result values, so I really like this approach
I'll keep it mind